### PR TITLE
platform: Bump wpebackend-fdo requirement to 1.12

### PR DESCRIPTION
**Pull request by aperezdc**
Bump required wpebackend-fdo version to 1.12.0, which allows dropping conditional compilation of SHM buffer support and fullscreen handling code. Always building the code reduces the testing matrix, and version 1.12.0 is already over two years old so the change should not pose any issue for packagers.

### CODE REVIEW

**Comment by aperezdc** [comment]
Ping reviewers 🏓

File: platform/wayland/cog-utils-wl.h [hunk 1]
```diff
@@ -32,14 +32,6 @@ G_BEGIN_DECLS
 #    define HAVE_REFRESH_RATE_HANDLING 0
 #endif
 
-#if defined(WPE_FDO_CHECK_VERSION)
```
Comment (psaavedra) [inline platform/wayland/cog-utils-wl.h:35]: Dont you have to do the very similar thing for the cog-platform-drm.c?:

``` c
platform/drm/cog-platform-drm.c-53-#if defined(WPE_FDO_CHECK_VERSION)
platform/drm/cog-platform-drm.c:54:# define HAVE_SHM_EXPORTED_BUFFER WPE_FDO_CHECK_VERSION(1, 9, 0)
platform/drm/cog-platform-drm.c-55-#else
platform/drm/cog-platform-drm.c:56:# define HAVE_SHM_EXPORTED_BUFFER 0
platform/drm/cog-platform-drm.c-57-#endif
```

File: platform/wayland/cog-utils-wl.h [hunk 2]
```diff
@@ -32,14 +32,6 @@ G_BEGIN_DECLS
...
-#    define HAVE_SHM_EXPORTED_BUFFER WPE_FDO_CHECK_VERSION(1, 9, 0)
-#    define HAVE_FULLSCREEN_HANDLING WPE_FDO_CHECK_VERSION(1, 11, 1)
```
Comment (psaavedra) [inline platform/wayland/cog-utils-wl.h:37]: Same here for `HAVE_FULLSCREEN_HANDLING`: 

``` sh
aperezdc/bump-fdo psaavedra@golpe ~/local/git/WPEBackend-fdo $ grep -C 3 -n -r HAVE_FULLSCREEN_HANDLING platform/
platform/gtk4/cog-platform-gtk4.c-32-#endif
platform/gtk4/cog-platform-gtk4.c-33-
platform/gtk4/cog-platform-gtk4.c-34-#if defined(WPE_FDO_CHECK_VERSION)
platform/gtk4/cog-platform-gtk4.c:35:#    define HAVE_FULLSCREEN_HANDLING WPE_FDO_CHECK_VERSION(1, 11, 1)
platform/gtk4/cog-platform-gtk4.c-36-#else
platform/gtk4/cog-platform-gtk4.c:37:#    define HAVE_FULLSCREEN_HANDLING 0
platform/gtk4/cog-platform-gtk4.c-38-#endif
platform/gtk4/cog-platform-gtk4.c-39-
platform/gtk4/cog-platform-gtk4.c-40-struct _CogGtk4PlatformClass {
--
platform/gtk4/cog-platform-gtk4.c-78-
platform/gtk4/cog-platform-gtk4.c-79-    int width;
platform/gtk4/cog-platform-gtk4.c-80-    int height;
platform/gtk4/cog-platform-gtk4.c:81:#if HAVE_FULLSCREEN_HANDLING
platform/gtk4/cog-platform-gtk4.c-82-    bool is_fullscreen;
platform/gtk4/cog-platform-gtk4.c-83-    bool waiting_fullscreen_notify;
platform/gtk4/cog-platform-gtk4.c-84-#endif
--
platform/gtk4/cog-platform-gtk4.c-241-                                                      win->device_scale_factor);
platform/gtk4/cog-platform-gtk4.c-242-}
platform/gtk4/cog-platform-gtk4.c-243-
platform/gtk4/cog-platform-gtk4.c:244:#if HAVE_FULLSCREEN_HANDLING
platform/gtk4/cog-platform-gtk4.c-245-static void
platform/gtk4/cog-platform-gtk4.c-246-dispatch_wpe_fullscreen_event(struct platform_window* win)
platform/gtk4/cog-platform-gtk4.c-247-{
--
platform/gtk4/cog-platform-gtk4.c-466-    return TRUE;
platform/gtk4/cog-platform-gtk4.c-467-}
platform/gtk4/cog-platform-gtk4.c-468-
platform/gtk4/cog-platform-gtk4.c:469:#if HAVE_FULLSCREEN_HANDLING
platform/gtk4/cog-platform-gtk4.c-470-static bool
platform/gtk4/cog-platform-gtk4.c-471-on_dom_fullscreen_request(void* unused, bool fullscreen)
platform/gtk4/cog-platform-gtk4.c-472-{
--
platform/gtk4/cog-platform-gtk4.c-583-    g_signal_connect(window->gl_drawing_area, "resize", G_CALLBACK(resize), window);
platform/gtk4/cog-platform-gtk4.c-584-    g_signal_connect(window->gl_drawing_area, "notify::scale-factor", G_CALLBACK(scale_factor_change), window);
platform/gtk4/cog-platform-gtk4.c-585-
platform/gtk4/cog-platform-gtk4.c:586:#if HAVE_FULLSCREEN_HANDLING
platform/gtk4/cog-platform-gtk4.c-587-    g_signal_connect(window->gtk_window, "notify::fullscreened", G_CALLBACK(on_fullscreen_change), window);
platform/gtk4/cog-platform-gtk4.c-588-#endif
platform/gtk4/cog-platform-gtk4.c-589-
--
platform/gtk4/cog-platform-gtk4.c-699-    setup_fdo_exportable(&win);
platform/gtk4/cog-platform-gtk4.c-700-    cog_gamepad_setup(gamepad_provider_get_view_backend_for_gamepad);
platform/gtk4/cog-platform-gtk4.c-701-
platform/gtk4/cog-platform-gtk4.c:702:#if HAVE_FULLSCREEN_HANDLING
platform/gtk4/cog-platform-gtk4.c-703-    wpe_view_backend_set_fullscreen_handler(webkit_web_view_backend_get_wpe_backend(win.view_backend),
platform/gtk4/cog-platform-gtk4.c-704-                                            on_dom_fullscreen_request, NULL);
platform/gtk4/cog-platform-gtk4.c-705-#endif
```

**Review by psaavedra — CHANGES_REQUESTED** [review]
I like the idea of simplify the complexity of the code buy dropping the conditional compilation. I'd like to extend the changes to the gtk4 and drm platforms too.

**Comment by aperezdc** [comment]
> I like the idea of simplify the complexity of the code buy dropping the conditional compilation. I'd like to extend the changes to the gtk4 and drm platforms too.

Good point! I will update the patch to simplify them as well.

**Comment by aperezdc** [comment]
Actually, we might want to update the requirement to 1.14, which is now a year and a half old... that would allow removing more conditionals 🤔

**Review by psaavedra — APPROVED** [review]

**Pull request merged**